gateway/gateway-runtime/policy-engine/internal/admin/server_test.go

```
@@ -356,3 +402,292 @@ func TestIPWhitelistMiddleware_PreservesRequestPath(t *testing.T) {
+	reg := &registry.PolicyRegistry{Policies: make(map[string]*registry.PolicyEntry)}
+
+	server := NewServer(cfg, k, reg, &mockXDSSyncProvider{version: "pc-v11"}, nil, nil)
+	require.NotNil(t, server.tlsServer)
+
+	ctx := context.Background()
+	errChan := make(chan error, 1)
+	go func() { errChan <- server.Start(ctx) }()
+	time.Sleep(100 * time.Millisecond)
+
+	// The plaintext listener is unaffected by enabling TLS.
+	plainResp, err := http.Get(fmt.Sprintf("http://127.0.0.1:%d/health", plainPort))
```

golangci-lint (2.13.2)

[error] net/http.Get must not be called. use net/http.NewRequestWithContext and (*net/http.Client).Do(*http.Request)

(noctx)